.github/actions/merge-base-branch/action.yml

```diff
@@ -0,0 +1,26 @@
+name: 'Merge from Base Branch if Action Running on a PR'
+description: 'Merge from the base branch the PR is being merged into.'
+runs:
+  using: 'composite'
+  steps:
+    - id: merge
+      name: Merge
+      run: |
+        if [[ ${{ github.event_name}} != "pull_request" ]]
+        then
+          echo "Skipping merge because not triggered by a PR event."
+          exit 0
+        fi
+        echo "Base Repository: ${{ github.repository }}"
+        echo "Base Branch: ${{ github.base_ref }}"
+        echo "Head Repository: ${{ github.event.pull_request.head.repo.full_name }}"
+        echo "Head Branch: $GITHUB_HEAD_REF"
+        git config --global user.email "admin@oppia.org"
+        git config --global user.name "Oppia Merge Base Branch GitHub Action"
+        git remote add source "https://github.com/${{ github.event.pull_request.head.repo.full_name }}.git"
+        git remote add base "https://github.com/${{ github.repository }}.git"
+        git fetch source $GITHUB_HEAD_REF
+        git fetch base ${{ github.base_ref }}
+        git checkout base/${{ github.base_ref }}
+        git merge source/$GITHUB_HEAD_REF
```

Comment:
In the container, we have the contributor's feature branch, right? (not sure) and we need to merge with `oppia/oppia develop`, right? OR is it the opposite? If the latter one, then how it is working as of now? [not sure!]

Reply:
In the container, we have a merge commit GitHub creates by merging the feature branch into `develop`. However, this merge commit doesn't get updated when we push commits to `develop`. With this PR, we stop using GitHub's merge commit and instead do the merge ourselves to ensure we get the latest `develop`.